pkg/datapath/linux/node_linux_test.go

```diff
@@ -1026,18 +1027,41 @@ func (s *linuxPrivilegedIPv4OnlyTestSuite) TestArpPingHandling(c *check.C) {
 	})
 	c.Assert(err, check.IsNil)
 
+	wait := func(now time.Time, nodeID nodeTypes.Identity, isDelete bool) {
+		err := testutils.WaitUntil(func() bool {
```

Comment:
From my reading of the code `now` is a lower bound for when the operation happened.

However, it is used only for insertion and not deletion (since for the latter, we do not have a deletion timestamp to check).
Given this, and the overall structure of the function, I wonder if it would be better to have two versions of this function one for deletes and one for inserts.

I think this would also improve readability on the caller side (at the cost of some code duplication).

Reply:
OK, sounds good.